packages/muelu/test/scaling/MMKernelDriver.cpp

```diff
@@ -403,107 +403,122 @@ void Multiply_KokkosKernels(const Xpetra::Matrix<Scalar,LocalOrdinal,GlobalOrdin
-    const LO colMapSize = static_cast<LO>(Bview.colMap->getNodeNumElements());
-    Kokkos::parallel_for("Tpetra::mult_A_B_newmatrix::Bcol2Ccol_fill",
-                         Kokkos::RangePolicy<execution_space, LO>(0, colMapSize),
-                         KOKKOS_LAMBDA(const LO i) {
-                           Bcol2Ccol(i) = i;
-                         });
-
-    // Acol2Brow
-    local_map_type Acolmap_local = Aview.colMap->getLocalMap();
-    local_map_type Browmap_local = Bview.origMatrix->getRowMap()->getLocalMap();
-    lo_view_t targetMapToOrigRow(Kokkos::ViewAllocateWithoutInitializing("targetMapToOrigRow"),Aview.colMap->getNodeNumElements());
-    lo_view_t targetMapToImportRow;
-    Kokkos::parallel_for("Tpetra::mult_A_B_newmatrix::construct_tables",range_type(Aview.colMap->getMinLocalIndex(), Aview.colMap->getMaxLocalIndex()+1),KOKKOS_LAMBDA(const LO i) {
-      GO aidx = Acolmap_local.getGlobalElement(i);
-      LO B_LID = Browmap_local.getLocalElement(aidx);
-      if (B_LID != LO_INVALID) {
-        targetMapToOrigRow(i)   = B_LID;
-        //        targetMapToImportRow(i) = LO_INVALID;
-      } else {
-        // This shouldn't happen here
-      }
-    });
-    tm = Teuchos::null;
-    Au->getComm()->barrier();
-
-    // **********************************
-    // Multiply
-    Teuchos::RCP<Teuchos::ParameterList> params;
-    tm = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("MM LTG: Multiply")));
-    Tpetra::MatrixMatrix::ExtraKernels::mult_A_B_newmatrix_LowThreadGustavsonKernel(Aview,Bview,targetMapToOrigRow,targetMapToImportRow,Bcol2Ccol,Icol2Ccol,*Cnc,Cimport,std::string("LTG Test"),params);
-
-    tm = Teuchos::null;
-    Au->getComm()->barrier();
-
-#endif
-
+    using Teuchos::RCP;
+    using Teuchos::rcp;
+    using Teuchos::TimeMonitor;
+
+    Xpetra::UnderlyingLib lib = A.getRowMap()->lib();
+    RCP<TimeMonitor> tm;
+
+    if (lib == Xpetra::UseTpetra) {
+      typedef Tpetra::CrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node> crs_matrix_type;
+      typedef Tpetra::Import<LocalOrdinal,GlobalOrdinal,Node>          import_type;
+      typedef typename crs_matrix_type::local_matrix_type    KCRS;
+      typedef typename KCRS::device_type device_t;
+      typedef typename KCRS::StaticCrsGraphType graph_t;
+      typedef typename graph_t::row_map_type::non_const_type lno_view_t;
+      typedef typename graph_t::entries_type::non_const_type lno_nnz_view_t;
+      typedef typename KCRS::values_type::non_const_type scalar_view_t;
+      typedef Kokkos::View<LO*, typename lno_view_t::array_layout, typename lno_view_t::device_type> lo_view_t;
+      typedef Tpetra::Map<LO,GO,NO>                                     map_type;
+      typedef typename map_type::local_map_type                         local_map_type;
+      typedef typename Node::execution_space execution_space;
+      typedef Kokkos::RangePolicy<execution_space, size_t> range_type;
+      LocalOrdinal LO_INVALID = Teuchos::OrdinalTraits<LO>::invalid();
```

Comment:
@mhoemmen I think since this is in MueLu and will never be put inside a CUDA kernel, I'll stick with the Teuchos version (the rest of MueLu also uses only Teuchos::OrdinalTraits).